framework/src/org/checkerframework/framework/flow/CFAbstractValue.java

```diff
@@ -4,550 +4,454 @@
 import org.checkerframework.checker.nullness.qual.Nullable;
 */
 
+import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Set;
 import javax.annotation.processing.ProcessingEnvironment;
 import javax.lang.model.element.AnnotationMirror;
-import javax.lang.model.type.IntersectionType;
 import javax.lang.model.type.TypeKind;
 import javax.lang.model.type.TypeMirror;
+import javax.lang.model.type.TypeVariable;
+import javax.lang.model.type.WildcardType;
 import javax.lang.model.util.Types;
 import org.checkerframework.dataflow.analysis.AbstractValue;
 import org.checkerframework.dataflow.qual.Pure;
 import org.checkerframework.dataflow.qual.SideEffectFree;
 import org.checkerframework.dataflow.util.HashCodeUtils;
-import org.checkerframework.framework.type.AnnotatedTypeFactory;
 import org.checkerframework.framework.type.AnnotatedTypeMirror;
-import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedArrayType;
-import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedIntersectionType;
 import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedTypeVariable;
-import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedWildcardType;
 import org.checkerframework.framework.type.GenericAnnotatedTypeFactory;
 import org.checkerframework.framework.type.QualifierHierarchy;
 import org.checkerframework.framework.type.TypeHierarchy;
-import org.checkerframework.framework.type.visitor.AnnotatedTypeMerger;
 import org.checkerframework.framework.util.AnnotatedTypes;
+import org.checkerframework.framework.util.PluginUtil;
 import org.checkerframework.javacutil.AnnotationUtils;
 import org.checkerframework.javacutil.ErrorReporter;
 import org.checkerframework.javacutil.InternalUtils;
+import org.checkerframework.javacutil.TypesUtils;
 
 /**
- * An implementation of an abstract value used by the Checker Framework org.checkerframework.dataflow
- * analysis. Contains a set of annotations.
+ * An implementation of an abstract value used by the Checker Framework
+ * org.checkerframework.dataflow analysis.
  *
  * @author Stefan Heule
- *
  */
 public abstract class CFAbstractValue<V extends CFAbstractValue<V>> implements AbstractValue<V> {
 
     /**
-     * The analysis class this store belongs to.
+     * The analysis class this value belongs to.
      */
     protected final CFAbstractAnalysis<V, ?, ?> analysis;
 
     protected final TypeHierarchy typeHierarchy;
-
-    /**
-     * The type (with annotations) corresponding to this abstract value.
-     */
-    protected final AnnotatedTypeMirror type;
-
-    public CFAbstractValue(CFAbstractAnalysis<V, ?, ?> analysis, AnnotatedTypeMirror type) {
-        assert AnnotatedTypes.isValidType(analysis.qualifierHierarchy, type)
-                : "Encountered invalid type: " + type.toString(true);
+    protected final GenericAnnotatedTypeFactory<V, ?, ?, ?> factory;
```

Comment:
The values used to store a reference to an ATM that has many more references than the few that  are here.  So, the memory consumption should be less than before.